pkg/proxy/kafka.go

```diff
@@ -356,6 +356,19 @@ func (k *kafkaRedirect) handleRequestConnection(pair *connectionPair) {
 	}), "Proxying request Kafka connection")
 
 	handleRequests(k.socket.closing, pair, pair.Rx, nil, k.handleRequest)
+
+	// The proxymap contains an entry with metadata for the receive side of the
+	// connection, remove it after the connection has been closed.
+	if pair.Rx != nil {
+		// We are running in our own go routine here so we can just
+		// block this go routine until after the connection is
+		// guaranteed to have been closed
+		time.Sleep(proxyConnectionCloseTimeout)
```

Comment:
I would add a small buffer, to let an eventual RST packet get through. In the worst case, the IP stack will decide to send an RST just around that time. We need to give it a chance to get through. How about adding like 1 second?
```
time.Sleep(proxyConnectionCloseTimeout + 1 * time.Second)
```

Reply:
Done, can't hurt. It's probably unlikely though that we can transmit an RST if we can't flush out the data and close the connection within the timeout. It's likely a connectivity problem anyway.